File: Minimal/Dart/url_shortener.dart
import 'dart:math';

class URLShortener {
  Map<String, String> _urls = {};
  
  String shorten(String url) {
    String short = _generateShortCode(6);
    _urls[short] = url;
    return short;
  }
  
  String? getUrl(String short) {
    return _urls[short];
  }
  
  String _generateShortCode(int length) {
    const String characters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
    Random random = Random();
    String short = '';
    
    for (int i = 0; i < length; i++) {
      short += characters[random.nextInt(characters.length)];
    }
    
    return short;
  }
}

// Example usage
void main() {
  URLShortener shortener = URLShortener();
  String short = shortener.shorten("https://www.example.com");
  print("Short URL: $short");
  String? original = shortener.getUrl(short);
  print("Original URL: $original");
}

File: Minimal/Lua/url_shortener.lua
-- URLShortener class
URLShortener = {}
URLShortener.__index = URLShortener

function URLShortener:new()
    local obj = {
        urls = {}
    }
    setmetatable(obj, URLShortener)
    return obj
end

function URLShortener:shorten(url)
    local short = self:generateShortCode(6)
    self.urls[short] = url
    return short
end

function URLShortener:getUrl(short)
    return self.urls[short]
end

function URLShortener:generateShortCode(length)
    local characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
    local short = ""
    
    math.randomseed(os.time())
    
    for i = 1, length do
        local index = math.random(1, #characters)
        short = short .. string.sub(characters, index, index)
    end
    
    return short
end

-- Example usage
local shortener = URLShortener:new()
local short = shortener:shorten("https://www.example.com")
print("Short URL: " .. short)
local original = shortener:getUrl(short)
print("Original URL: " .. (original or "nil"))
